agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/context/AIAgentContext.kt

```diff
@@ -119,4 +167,30 @@ internal class AIAgentContext(
         strategyId = strategyId ?: this.strategyId,
         pipeline = pipeline ?: @OptIn(InternalAgentsApi::class) this.pipeline,
     )
-}
\ No newline at end of file
+
+    /**
+     * Creates a copy of the current [AIAgentContext] with deep copies of all mutable properties.
+     *
+     * @return A new instance of [AIAgentContext] with copies of all mutable properties.
+     */
+    override suspend fun fork(): AIAgentContextBase = copy(
+        llm = this.llm.copy(),
+        storage = this.storage.copy(),
+        stateManager = this.stateManager.copy(),
+    )
+
+    /**
+     * Replaces the current context with the provided context.
+     * This method is used to update the current context with values from another context,
+     * particularly useful in scenarios like parallel node execution where contexts need to be merged.
+     *
+     * @param context The context to replace the current context with.]]
+     */
+    override suspend fun replace(context: AIAgentContextBase) {
```

Comment:
There's a potential problem with this method that might confuse someone (it confused me initially, at least). From the method signature and according to KDocs it looks like the whole context will be replaced, but in fact it's not the case and only three properties are actually replaced.